src/main.ts

```diff
@@ -101,29 +95,124 @@ async function handleFileSelect(event: any) {
+
+    let layerName = document.createElement('div');
+    layerName.textContent = layer.name;
+    layerName.setAttribute('id', 'layerName');
+
+    let flexRow = document.createElement('div');
+    flexRow.setAttribute('class', 'flex-row');
+    flexRow.setAttribute('style', `gap: 1px;`);
+
+    let layerCanvas = document.createElement('canvas');
+    layerCanvas.setAttribute('id', `layer-canvas${layerId}`);
+    layerCanvas.setAttribute('width', '200');
+    layerCanvas.setAttribute('height', '120');
+
+    layerCanvas.addEventListener('click', function () {
+        const isSelected = selectedLayers.includes(layerId);
+        handleLayerSelected(!isSelected);
+        updateCanvas();
+    });
+
+    let myCheckbox = document.createElement('input');
+    myCheckbox.setAttribute('type', 'checkbox');
+    myCheckbox.setAttribute('id', 'myCheckbox');
+    myCheckbox.checked = selectedLayers.includes(layerId) ? true : false;
+    myCheckbox.addEventListener('change', function () {
+        handleLayerSelected(myCheckbox.checked);
+        updateCanvas();
+    });
+
+    flexRow.appendChild(layerCanvas);
+    flexRow.appendChild(myCheckbox);
+    layerTile.appendChild(layerName);
+    layerTile.appendChild(flexRow);
+    document.body.appendChild(layerTile);
+    parentDiv?.appendChild(layerTile);
+}
+
+function renderLayer(layer: Layer, canvas: HTMLCanvasElement, scaleFactor = 1) {
+    const ctx = canvas.getContext('2d') as CanvasRenderingContext2D;
+
     ctx.clearRect(0, 0, canvas.width, canvas.height);
     ctx.strokeStyle = 'black';
     ctx.lineWidth = 1;
-    const listRef = dxfHandler.currentLayer.lines;
-    const trans_x = -1 * dxfHandler.currentLayer.minPoint[0];
-    const trans_y = -1 * dxfHandler.currentLayer.minPoint[1];
+    const listRef = layer.lines;
+    const trans_x = -1 * layer.minPoint[0];
+    const trans_y = -1 * layer.minPoint[1];
     let count = 0;
     let x1, y1, x2, y2;
+
+    // Find the maximum x and y values in the layer
+    let maxX = Math.max(...listRef.filter((_, index) => index % 2 === 0));
+    let maxY = Math.max(...listRef.filter((_, index) => index % 2 !== 0));
+
+    // Calculate the scale factors for the x and y dimensions
+    let scaleX = canvas.width / maxX;
+    let scaleY = canvas.height / maxY;
+
+    // Use the smaller scale factor to avoid stretching
+    let scale = Math.min(scaleX, scaleY);
+
     while (count < listRef.length) {
         // Draw the line
         x1 = listRef[count];
         y1 = listRef[count + 1];
         x2 = listRef[count + 2];
         y2 = listRef[count + 3];
         ctx.beginPath();
-        ctx.moveTo((x1 + trans_x) * scaleFactor, (y1 + trans_y) * scaleFactor);
-        ctx.lineTo((x2 + trans_x) * scaleFactor, (y2 + trans_y) * scaleFactor);
+        ctx.moveTo((x1 + trans_x) * scale * scaleFactor, (y1 + trans_y) * scale * scaleFactor);
+        ctx.lineTo((x2 + trans_x) * scale * scaleFactor, (y2 + trans_y) * scale * scaleFactor);
```

Comment:
eehmm this seems weird, why are you scaling twice? :thinking:

Reply:
one is the adjustment for the size which ensures that by default its scaled to either height or width of the canvas. The other is the factor you can apply on top, if the scaling by default does not fit (happens because of wide range between min and max).